File: fixtures/scale/ruby/large/lib/check_foldsum.rb
require_relative 'foldsum_digest'
require_relative 'foldsum_digest_pair'
require_relative 'corpus_hash'

def check_foldsum
  corpus_hash.each do |text|
    first = foldsum_digest(text)
    second = foldsum_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_foldsum.py
from cipherlab.ciphers.foldsum import foldsum_digest
from cipherlab.specs.foldsum_spec import foldsum_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_foldsum():
    spec = foldsum_spec()
    for text in corpus_hash():
        first = foldsum_digest(text)
        second = foldsum_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
